core/unit_test/TestGraph.hpp

```diff
@@ -329,32 +329,38 @@ TEST_F(TEST_CATEGORY_FIXTURE(graph), when_all_cycle) {
   //----------------------------------------
 }
 
-// This test is disabled because we don't currently support copying to host,
+// This test requires that the graph execution space can access
+// the host memoy space because we don't currently support copying to host,
 // even asynchronously. We _may_ want to do that eventually?
-TEST_F(TEST_CATEGORY_FIXTURE(graph), DISABLED_repeat_chain) {
-  auto graph = Kokkos::Experimental::create_graph(
-      ex, [&, count_host = count_host](auto root) {
-        //----------------------------------------
-        root.then_parallel_for(1, set_functor{count, 0})
-            .then_parallel_for(1, count_functor{count, bugs, 0, 0})
-            .then_parallel_for(1, count_functor{count, bugs, 1, 1})
-            .then_parallel_reduce(1, set_result_functor{count}, count_host)
-            .then_parallel_reduce(
-                1, set_result_functor{bugs},
-                Kokkos::Sum<int, Kokkos::HostSpace>{bugs_host});
-        //----------------------------------------
-      });
+TEST_F(TEST_CATEGORY_FIXTURE(graph), repeat_chain) {
+  constexpr bool skip = !Kokkos::SpaceAccessibility<
+      Kokkos::Impl::remove_cvref_t<decltype(ex)>,
+      typename Kokkos::Impl::remove_cvref_t<
+          decltype(bugs_host)>::memory_space>::accessible;
```

Comment:
Why are we not just saying "host"?

Reply:
What do you mean ?

Reply:
that I don't understand why we have such a complex deduction of the space if in the end you know/you rely on the fact that this is the host.  I am basing this on the skip message you provided.

Reply:
I think I should be more specific in the message:
> The reduction target 'bugs_host' memory space cannot be accessed by the graph.

I just want to make sure that if someone changes the `bugs_host` to be in a shared space, this check will enable the test.